test.js

```diff
@@ -1538,6 +1538,29 @@ function runTests(baseopts) {
             done();
           });
       });
+      it('should emit an unlink event when a file is updated and deleted just after that', function(done) {
+        var spy = sinon.spy();
+        var testPath = getFixturePath('subdir/add.txt');
+        var filename = sysPath.basename(testPath);
+        options.cwd = sysPath.dirname(testPath);
+        fs.mkdir(options.cwd, w(function() {
+          fs.writeFile(testPath, 'hello', w(function() {
+            stdWatcher()
+              .on('all', spy)
+              .on('ready', function() {
+                fs.writeFile(testPath, 'edit', simpleCb);
+                w(function() {
```

Comment:
> Shouldn't this be moved to be the callback after the writeFile above?

I built this test case by trying to mimic what I saw in the logs where the error was found, and there was a very short delay between the write and the delete. But I don't think it's relevant here. Doing as you said you should work too, and is probably simpler to understand.

> Also, overriding the timeout to 10ms makes detecting the lack of a change event not very meaningful.

Yes, you are right. The `change` event won't appear before the `unlink` one in this test, and should be tested later in the test.

> The default time, or anything that gives it some time to react while still within the awf stability threshold would be better.

I'm not sure to understand what you have in mind. You have a lot more experience than me at testing chokidar. So, feel free to refactor this test!